Tasks/GooglePlayStatusUpdate/googleutil.ts

```diff
@@ -0,0 +1,256 @@
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @return {Promise} edit - A promise that will return result from inserting a new edit
+ *                          { id: string, expiryTimeSeconds: string }
+ */
+export async function getNewEdit(edits: any, globalParams: GlobalParams, packageName: string): Promise<Edit> {
+    tl.debug('Creating a new edit');
+    const requestParameters: PackageParams = {
+        packageName: packageName
+    };
+
+    tl.debug('Additional Parameters: ' + JSON.stringify(requestParameters));
+    const res = await edits.insert(requestParameters);
+    return res.data;
+}
+
+/**
+ * Gets information for the specified app and track
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @param {string} track - one of the values {"internal", "alpha", "beta", "production"}
+ * @returns {Promise} track - A promise that will return result from updating a track
+ *                            { track: string, versionCodes: [integer], userFraction: double }
+ */
+export async function getTrack(edits: any, packageName: string, track: string): Promise<Track> {
+    tl.debug('Getting Track information');
+    const requestParameters: PackageParams = {
+        packageName: packageName,
+        track: track
+    };
+
+    tl.debug('Additional Parameters: ' + JSON.stringify(requestParameters));
+    const getTrack = await edits.tracks.get(requestParameters);
+    return getTrack.data;
+}
+
+/**
+ * Update a given release track with the given information
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @param {string} track - one of the values {"internal", "alpha", "beta", "production"}
+ * @param {integer or [integers]} versionCode - version code returned from an apk call. will take either a number or a [number]
+ * @param {string} status - one of the values {"draft", "inProgress", "halted", "completed"}
+ * @param {double} userFraction - for rollouting out a release to a track.
+ * @param {releaseNotes} releaseNotes - optional release notes to be attached as part of the update
+ * @returns {Promise} track - A promise that will return result from updating a track
+ *                            { track: string, versionCodes: [integer], userFraction: double }
+ */
+export async function updateTrack(edits: any, packageName: string, track: string, versionCode: any, status: string, userFraction: number, releaseNotes?: ReleaseNotes[]): Promise<Track> {
+    tl.debug('Updating track');
+    const release: AndroidRelease = {
+        versionCodes: (typeof versionCode === 'number' ? [versionCode] : versionCode)
+    };
+
+    if (releaseNotes && releaseNotes.length > 0) {
+        tl.debug('Attaching release notes to the update');
+        release.releaseNotes = releaseNotes;
+    }
+
+    if (!Number.isNaN(userFraction)) {
+        release.userFraction = userFraction;
```

Comment:
Will it be expected for customer if non a number value will be ignored? Could it make sense to throw the error for this case?

Reply:
Added judgment on userFraction, illegal input such as non-empty non-number throws an error.